ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -175,6 +175,9 @@ let code_action store (params : Lsp.CodeAction.Params.t) =
             (code_action_of_case_analysis params.textDocument.uri res)
         ]
       with
+      | Failure _
+      (* TODO: find a better way of catching this, this is a bit dirty *)
+      | Invalid_argument _
```

Comment:
I would prefer if you just changed our vendored copy of merlin to raise a new exception. Invalid_argument can be raised by quite a few different calls, so it's too dangerous to catch it here.

Reply:
That is better indeed ! 
What's the workflow there ? Should I submit a PR to the merlin repo first ?

Reply:
Submit a PR to my fork: https://github.com/rgrinberg/merlin/tree/remove-lsp

Submitting upstream is also good if you think your solution is good enough. I usually rebase that branch from time to time periodically.